### PR TITLE
Add rides to park show page

### PR DESCRIPTION
**Pull request by ethanysc**

### CODE REVIEW

File: app/controllers/api/v1/amusement_parks_controller.rb [hunk 1]
```diff
@@ -1,9 +1,23 @@
-class Api::V1::AmusementParksController < ApplicationController
+class Api::V1::AmusementParksController < ApiController
+  protect_from_forgery unless: -> {request.format.json?}
```
Comment (dpickett) [inline app/controllers/api/v1/amusement_parks_controller.rb]: this can go and only needs to be invoked in the `ApiController`

File: app/controllers/api/v1/amusement_parks_controller.rb [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
-    render json: AmusementPark.find(params[:id])
+    render json: {
+      amusement_park: AmusementPark.find(params[:id]),
+      reviews: serialized_reviews,
+      rides: serialized_rides
+    }
+  end
+
+  def serialized_reviews
+    ActiveModel::Serializer::ArraySerializer.new(AmusementPark.find(params[:id]).reviews, each_serializer: ReviewSerializer)
```
Comment (dpickett) [inline app/controllers/api/v1/amusement_parks_controller.rb]: what happens if the `AmusementPark` is not found? Also, could you use the one that is already found in the body of the `show`?

File: app/controllers/api/v1/reviews_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Api::V1::ReviewsController < ApiController
```
Comment (dpickett) [inline app/controllers/api/v1/reviews_controller.rb]: reviews should not be part of this PR.

File: app/controllers/api/v1/rides_controller.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      features: serialized_features
+      }
+  end
+
+  def serialized_ride
+    ActiveModel::Serializer.new(Ride.find(params[:id]), each_serializer: RideSerializer)
+  end
+
+  def serialized_features
+    ActiveModel::Serializer::ArraySerializer.new(Ride.find(params[:id]).ride_features, each_serializer: RideFeatureSerializer)
```
Comment (dpickett) [inline app/controllers/api/v1/rides_controller.rb]: what happens if the `Ride` is not found?

**Comment by dpickett** [comment]
let's merge #20 before we tackle what needs to be done here

File: app/controllers/api/v1/rides_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    def show
+    # amusement_park = AmusementPark.find(params[:id])
+    # payload = {
+    #   park: amusement_park,
+    #   reviews: amusement_park.reviews
+    # }
+    # render json: payload
+
+    # render json: AmusementPark.find(params[:id])
+    # render json: AmusementPark.find(params[:id]).reviews
```
Comment (bkincart) [inline app/controllers/api/v1/rides_controller.rb]: Please remove commented code before merging

File: app/controllers/api/v1/rides_controller.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    
+    render json: {
+      amusement_park_id: AmusementPark.find(params[:amusement_park_id]).id,
+      ride: Ride.find(params[:id]),
+      features: serialized_features
+      }
```
Comment (bkincart) [inline app/controllers/api/v1/rides_controller.rb]: Un-indent this line

File: app/controllers/rides_controller.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class RidesController < ApplicationController
+
+  def show
+  end
```
Comment (bkincart) [inline app/controllers/rides_controller.rb]: Please remove whitespace above and below. While you're at it, you can make this one line as so:
```
def show; end
```

File: app/javascript/react/containers/AmusementParksShowContainer.js [hunk 8]
```diff
@@ -63,7 +66,7 @@ class AmusementParksShowContainer extends React.Component {
   }
 
   render(){
-    let { amusementPark, reviews } = this.state
+    let { amusementPark, reviews, rides } = this.state
```
Comment (bkincart) [inline app/javascript/react/containers/AmusementParksShowContainer.js]: Really awesome object destructuring!

File: app/views/rides/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<div id="app">
+
+</div>
```
Comment (bkincart) [inline app/views/rides/show.html.erb]: Ideally this would be self-closing: `<div id='app' />`

File: coverage/index.html [hunk 10]
```diff
@@ -0,0 +1,80 @@
+<!doctype html>
```
Comment (bkincart) [inline coverage/index.html]: Please add `/coverage/*` to your `.gitignore` file and run `git rm coverage/index.html` and so on for each of the files showing up here under `coverage` folder

File: spec/javascript/exampleTest.js [hunk 11]
```diff
@@ -20,53 +27,89 @@ describe('Amusement Parks Show', () => {
...
+  // make a test to assert the initial state is as expect
+
   describe('AmusementParksShowContainer', () => {
 
-    it('render the parks name', (done) => {
+    // make assertion that a tile is present
+    // make assertion that the tile is receiving the right props
+    // rinse and repeat for each tile
+
+    fit('ParkShowTile is present and receiving the correct props', (done) => {
```
Comment (bkincart) [inline spec/javascript/exampleTest.js]: Please remove focus `f` on this test

File: spec/javascript/exampleTest.js [hunk 12]
```diff
@@ -20,53 +27,89 @@ describe('Amusement Parks Show', () => {
...
+          operating_season: 'April through late December',
+          website: 'https://www.sixflags.com/newengland'
+        });
         done()
 
       }, 0)
     })
 
-    it('render the parks address', (done) => {
+    fit('ReviewsContainer is present and receiving the correct props', (done) => {
```
Comment (bkincart) [inline spec/javascript/exampleTest.js]: Remove focus here too

File: spec/javascript/exampleTest.js [hunk 13]
```diff
@@ -20,53 +27,89 @@ describe('Amusement Parks Show', () => {
...
+        expect(wrapper.find(ReviewsContainer).props()).toEqual({
+          reviews: []
+        });
         done()
 
       }, 0)
     })
 
-    it('render the parks city', (done) => {
+    fit('ReviewFormContainer is present and receiving the correct props', (done) => {
```
Comment (bkincart) [inline spec/javascript/exampleTest.js]: And here

File: spec/javascript/exampleTest.js [hunk 14]
```diff
@@ -20,53 +27,89 @@ describe('Amusement Parks Show', () => {
...
+
+        expect(wrapper.find(ReviewFormContainer)).toBePresent();
+
         done()
 
       }, 0)
     })
 
-    it('render the parks zipcode', (done) => {
+    fit('RidesIndexContainer is present and receiving the correct props', (done) => {
```
Comment (bkincart) [inline spec/javascript/exampleTest.js]: Finally here
